wire-ios-data-model/Source/Model/Conversation/ZMConversation+SelfConversation.swift

```diff
@@ -78,54 +80,93 @@ extension ZMConversation {
+        let selfConversation = ZMConversation.selfConversation(in: context)
         return try selfConversation.appendClientMessage(with: message, expires: false, hidden: false)
     }
 
-    static func updateConversation(withLastReadFromSelfConversation lastRead: LastRead, inContext moc: NSManagedObjectContext) {
-        let newTimeStamp = Double(integerLiteral: lastRead.lastReadTimestamp)
-        let timestamp = Date(timeIntervalSince1970: newTimeStamp/1000)
+    private static func sendMessageOverMLSSelfConversation(
+        _ content: MessageCapable,
+        in context: NSManagedObjectContext
+    ) throws -> ZMClientMessage? {
+        guard let selfConversation = ZMConversation.fetchSelfMLSConversation(in: context) else {
+            return nil
+        }
+
+        let message = GenericMessage(content: content, nonce: UUID())
+        return try selfConversation.appendClientMessage(with: message, expires: false, hidden: false)
+    }
+
+    // MARK: - Sync downstream
+
+    static func updateConversation(
+        withLastReadFromSelfConversation lastRead: LastRead,
+        in context: NSManagedObjectContext
+    ) {
         guard let conversationID = UUID(uuidString: lastRead.conversationID) else {
             return
         }
-        let conversation = ZMConversation.fetchOrCreate(with: conversationID, domain: lastRead.qualifiedConversationID.domain, in: moc)
+
+        let newTimeStamp = Double(integerLiteral: lastRead.lastReadTimestamp)
+        let timestamp = Date(timeIntervalSince1970: newTimeStamp/1000)
+        let conversation = ZMConversation.fetchOrCreate(
+            with: conversationID,
+            domain: lastRead.qualifiedConversationID.domain,
+            in: context
+        )
         conversation.updateLastRead(timestamp, synchronize: false)
     }
 
-    static func updateConversation(withClearedFromSelfConversation cleared: Cleared, inContext moc: NSManagedObjectContext) {
-        let newTimeStamp = Double(integerLiteral: cleared.clearedTimestamp)
-        let timestamp = Date(timeIntervalSince1970: newTimeStamp/1000)
+    static func updateConversation(
+        withClearedFromSelfConversation cleared: Cleared,
+        in context: NSManagedObjectContext
+    ) {
         guard let conversationID = UUID(uuidString: cleared.conversationID) else {
             return
         }
-        let conversation = ZMConversation.fetchOrCreate(with: conversationID, domain: cleared.qualifiedConversationID.domain, in: moc)
-        conversation.updateCleared(timestamp, synchronize: false)
 
+        let newTimeStamp = Double(integerLiteral: cleared.clearedTimestamp)
+        let timestamp = Date(timeIntervalSince1970: newTimeStamp/1000)
+        let conversation = ZMConversation.fetchOrCreate(
+            with: conversationID,
+            domain: cleared.qualifiedConversationID.domain,
+            in: context
+        )
+        conversation.updateCleared(timestamp, synchronize: false)
```

Comment:
can we factor common code with method above?

Reply:
done